### DISCUSSION TITLE
v59.0-alpha1

### DISCUSSION THREAD

**Discussion by westnordost** [streetcomplete/StreetComplete#5812] [Release Discussions]
### 🗺️ MapLibre (#5693), by @Helium314 and me

Migrated from the map renderer tangram-es to [MapLibre](https://maplibre.org/). The former has effectively been unmaintained for quite a few years and does not support modern iOS, this solves some long-standing display and crash issues.

Display-wise, nothing much *should* change, the map will just look and behave a little different.

Issues solved by the migration are #5601, #5241, #5216, #5161, #5146, #5137, #5048, #5027, #5006, #4844, #4830, #4556, #4554, #4550, #4542, #4522, #4277, #4259, #4026, #3417, #3397, #3346, #3275, #3022, #2878, #2711, #2703, #2612, #2571, #2373, #2359, #2358, #2187, #1713, #1684, #1264, #1019, #869, #807, #318, #179, [mapstyle#119](https://github.com/streetcomplete/streetcomplete-mapstyle/issues/119), [mapstyle#56](https://github.com/streetcomplete/streetcomplete-mapstyle/issues/56)... maybe more. Thanks to @riQQ for this list.

### General improvements

- Redesigned the settings screens, about screens and tutorial screens (#5719)
- minor improvements for right-to-left localization (Arabic, Hebrew, Farsi)
- fix rare crash issues (#5804)

### New Quests

- _"What is the width of the opening here?"_, asked for gates etc. (#5713, #5800) by @Nick-Tallguy

### Removed Quests

- Asking _"Is this a one-way street?"_, based on processed data from ImproveOSM because Grab is shutting it down (#5725)
- Asking _"Can you pick up parcels here?"_ and _"Can you drop off parcels here?"_ because it turns out that it is often not surveyable and the answer is always the same for one brand of parcel lockers (#5763)

### Overlay Improvements

- Addresses: Presence of housenumbers is now highlighted in blue (#5748)
- Bike paths, Sidewalks: Highlight less situations as missing (#5716, #5722)
- Bike paths: When selecting "designated shared-use path", always mark it as designated for pedestrians, too (#5742)
- allow hiding notes that block editing with an overlay (#5703)

### Quest Improvements

- Ask many road-related quests also for busways (#5749)
- Recycling containers: Also ask for containers that have been mapped as an area (#5735)
- Recycling containers: Also highlight nearby waste containers and bins (#5754)
- Accepting cards: Don't ask for places that explicitly require no fee (#5772), by @HolgerJeromin
- Drinking water: Don't ask for intermittent water sources (#5750), by @mnalis
- Track type: Don't ask for track type on bridges (#5731), by @kmpoppe
- Building levels, roof shapes: Also ask for houseboats (#5737), by @HolgerJeromin
- Traffic signals vibration: Disabled in Finland (#5795), by @mnalis
- Clothing container operators: Allow answering that there's no information on the container (#5764)
- Check existence: Don't ask if something else on the same place has just been checked (#5729)
- Other small improvements (#5740, #5792, #5548, #5776, ...), thanks @mnalis

<hr /><em>This discussion was created from the release <a href='https://github.com/streetcomplete/StreetComplete/releases/tag/v59.0-alpha1'>v59.0-alpha1</a>.</em>

**Comment by IdeallyGrey** [comment]
Just curious, you mention the old map renderer no longer supports modern iOS, yet StreetComplete is for Android. Is iOS support a future goal?

**Reply by HolgerJeromin** [reply]
Yes. See #5421

**Reply by westnordost** [reply]
To be precise, the map renderer [tangram-es](https://github.com/tangrams/tangram-es) **does** work on iOS, but uses the _OpenGL ES_ graphics API to render the map. Now, since a couple of years, Apple declared OpenGL ES on iOS deprecated, developers should rather use their own graphics API called _Metal_. So, while apps that use tangram-es for map rendering may still run on modern iOS devices, it is unclear for how much longer. Meanwhile, even Android is replacing _OpenGL ES_ with _Vulkan_ in the coming years.

So, the migration TBH has **primarily** been done because tangram-es is just not maintained anymore, bugs are not fixed, the library is not adapted to modern standards etc. but **also** because it is another step to eventually reach an iOS port of StreetComplete.

MapLibre-native does support Metal, and Vulkan support is being added. The project is in very active development and basically the open source industry standard when it comes down to displaying vector tiled maps (based on OSM data).

**Comment by ERYpTION** [comment]
Would a quest similar to the new "What is the width of the opening here?" be possible for barrier=cycle_barrier regarding the passableness of cargo bikes, wheelchairs etc? Or is it too difficult to judge whether they can pass since it often does not simply depend on max width?

**Reply by matkoniecz** [reply]
maybe at least for simple `barrier=cycle_barrier` with single obvious gap width it would be doable?

**Reply by Nick-Tallguy** [reply]
[Wiki for cycle_barriers](https://wiki.openstreetmap.org/wiki/Tag:barrier%3Dcycle_barrier#Passableness_for_cargo_bikes,_trailers,_wheelchairs,_etc.)
I'm interested in this for the future.

**Reply by mnalis** [reply]
- https://github.com/streetcomplete/StreetComplete/pull/5820

**Comment by RubenKelevra** [comment]
@westnordost I guess this means we need new project screenshots for the readme. Should I give that a go? 🤔

**Reply by matkoniecz** [reply]
website/android play preview also may need new ones, in multiple languages

**Reply by westnordost** [reply]
Thank you, but oh, not that much changed yet. And it is a lot of work, because I did create them for many different languages. It also needs to be done with a specific (virtual) device to fit the frame displayed on https://streetcomplete.app/

I think this should rather be done later, when I also ported the main screen controls and other things visible in the screenshots, e.g. the quest bubble ball pit will need to be replaced with something different, see #5595

**Reply by RubenKelevra** [reply]
I gave it a go on my phone. But it's kinda hard to find the same content.

But now reading this, I don't think you can use them :D

Btw: I think it looks great on  https://streetcomplete.app/ with the phone around, but it looks odd to have the status bar on top and the “pull up” bar on the bottom in the readme - kinda like we don't know how to crop a screenshot ;)

https://github.com/streetcomplete/StreetComplete/pull/5825

**Reply by westnordost** [reply]
Yeah, sorry, I can't use them. Thank you for your effort though.

If I remember correctly, a few years ago I made them with a very specific Android emulator, I think Pixel6. When I re-do them (or someone else re-does them), probably it is best to use also a Pixel6 (emulator).

**Comment by RubenKelevra** [comment]
Just wanna say thank you. The panning, zooming is now so(!) much more smooth on my 3 yo midrange smartphone. Great job!

**Reply by HolgerJeromin** [reply]
Same here. 
I expected a buggy, unstable alpha but it just works so much smoother and better than the old version!

**Reply by westnordost** [reply]
Thank you for testing the alpha, guys! I didn't expect anyone would download and test it (apart from the devs)

**Reply by Helium314** [reply]
> The panning, zooming is now so(!) much more smooth on my 3 yo midrange smartphone

Interestingly on my 10 year old midrange phone I had the opposite impression (with an older MapLibre version though)

**Reply by davidpnewton** [reply]
As I said on the SCEE discussion area, the transition with the alpha has been extremely smooth for me. No crashes, and only minor hiccoughs in the form of the weird missing underground fire hydrant icon problem and the overlay icons in the things overlay showing up with grey backgrounds rather than white. Both now fixed.

Clearly it's been a lot of hard work over the last few months. It is appreciated.

**Comment by RubenKelevra** [comment]
There's been a report of rendering issues in of the caron in response to the alph, but it sounds like the issues it not related to the alpha itself but being there for a while now, if I understand him correctly:

<details>
<summary>Screenshots</summary>

![Screenshot_2024-08-23-12-03-32-834-edit_com twitter android](https://github.com/user-attachments/assets/a67b809a-4a81-4fb1-9fb0-a1dc893494f6)

![20240823_120346](https://github.com/user-attachments/assets/90622504-21ac-481f-83d0-2a6806964310)
![20240823_120348](https://github.com/user-attachments/assets/02e79685-b661-4e30-8b9f-f286d7dcf036)
![20240823_120350](https://github.com/user-attachments/assets/dc45608d-8add-4278-9ab9-5deb16503e2b)

</details>

I've asked him to create a bug report dedicated to this, just in case he doesn't pick up on that: can someone confirm this? 

Source: https://x.com/rudolfs58991170/status/1826240104991760839

**Reply by matkoniecz** [reply]
sounds like https://github.com/streetcomplete/StreetComplete/issues/5006

**Reply by matkoniecz** [reply]
though https://github.com/streetcomplete/StreetComplete/issues/5006#issuecomment-1540376174 is now confusing 0 has they tested with MapLibre?

**Reply by RubenKelevra** [reply]
Not sure, but the alpha looks fine to me. But I'm not sure if it was broken before - don't see a lot of carons in Germany

**Reply by mnalis** [reply]
I've added comment to https://github.com/streetcomplete/StreetComplete/issues/5006#issuecomment-2307009403 - yes, display of those carons and similar is broken on Tangram-ES (on Android 13 and above only) and work fine in MapLibre. So, it should fix their problem automatically when 59.0 gets released.

**Comment by RubenKelevra** [comment]
There seems to be some issues at the app start. SC just crashed 3 times in startup without even showing, forth time worked fine and is stable now.

**Reply by matkoniecz** [reply]
see https://github.com/streetcomplete/StreetComplete/issues/5829#event-13992730855

**Comment by HolgerJeromin** [comment]
Just as a note:
The just released alpha2 shows alpha1 in about screen :-)

**Reply by westnordost** [reply]
What? How is that possible?

**Reply by Nick-Tallguy** [reply]
And in changesets.  

https://www.openstreetmap.org/changeset/155897427

For me it's working well and looking really good - thanks to everyone involved in making it so.

**Reply by mnalis** [reply]
> What? How is that possible?

I don't know, but I can confirm that [v59.0-alpha2](https://github.com/streetcomplete/StreetComplete/releases/tag/v59.0-alpha2) shows `v59.0-alpha1` in `About` screen under `Version`. Clicking on it shows changelog which talks about `v59.0-alpha2`, though. Strange...

**Reply by HolgerJeromin** [reply]
Same in alpha 3 btw.
Android app info shows 3, about screen shows 1, changelog shows 2 (which is kind of expected)

**Reply by Helium314** [reply]
> What? How is that possible?

The text is set to `"v" + BuildConfig.VERSION_NAME`, and version name is clearly updated. So... not possible. If you Ctrl+B on `VERSION_NAME` you even see the correct version.

Maybe there is some sort of cache involved. For me even the compose preview shows the wrong number.

**Reply by mnalis** [reply]
> Maybe there is some sort of cache involved

Quite possible, IMHO.
Because, if I let GitHub build debug version of that `v59.0-alpha3` tag [for me](https://github.com/mnalis/StreetComplete/actions/runs/10648078669), then it will identify correctly as  `v59.0-alpha3` in `About` screen, unlike the [official release](https://github.com/streetcomplete/StreetComplete/releases/tag/v59.0-alpha3) version which still identifies as  `v59.0-alpha1` in he `About` screen...

**Reply by matkoniecz** [reply]
> What? How is that possible?

"There are only two hard things in Computer Science; cache invalidation and naming things and one-off errors" ?

I have seen some weird caching issues in Android Studio at times. At some points I ended deleteting repo (except .git) and doing git reset to force AS into noticing that files changed (it kept building my personal fork version partially with normal version release id, what was breaking photo-taking...).

**Reply by mnalis** [reply]
FYI, [v59.0-alpha4](https://github.com/streetcomplete/StreetComplete/releases/tag/v59.0-alpha4) now identifies itself correctly in `About` screen...
